src/main/java/com/foundrysoftware/minecraft/plugins/PluginTemplate.java

```diff
@@ -1,37 +1,171 @@
-        if (label.equalsIgnoreCase("simple")){
-            if (sender instanceof Player){
-                Player me = (Player)sender;
-                me.sendMessage("Plugin Working");
+        String simple="simple";
+        if (label.equalsIgnoreCase(simple)) {
+            if (sender instanceof Player) {
+                Player me = (Player) sender;
+                String message = "Plugin Working";
+                me.sendMessage(message);
                 return true;
             }
+            if (label.equalsIgnoreCase("hawkeye")) {
+                if (args[0].equalsIgnoreCase(ARROW_TYPE_TNT)) {
+                    if (sender instanceof Player) {
+                        Player me = (Player) sender;
+                        PlayerInventory inventory = me.getInventory();
+                        for (int i = 0; i < inventory.getSize(); i++) {
+                            ItemStack item = inventory.getItem(i);
+                            if (item == null) {
+                                continue;
+                            }
+                            if (isArrow(item) == true) {
+                                item.setType(Material.TIPPED_ARROW);
+                                item.setAmount(item.getAmount());
+
+                                String potion = "CONDUIT_POWER";
+
+                                item.setItemMeta(createPotion(item,potion));
+                            }
+                        }
+
+                    }
+                } else {
+                    if (sender instanceof Player) {
+                        Player me = (Player) sender;
+                        PlayerInventory inventory = me.getInventory();
+                        for (int i = 0; i < inventory.getSize(); i++) {
+                            ItemStack item = inventory.getItem(i);
+                            if (item == null) {
+                                continue;
+                            }
+                            if (isArrow(item) == true) {
+                                String potion = args[0];
+                                item.setType(Material.TIPPED_ARROW);
+                                item.setAmount(item.getAmount());
+
+                                item.setItemMeta(createPotion(item,potion));
+                            }
+
+                        }
+                    }
+                }
+            }
         }
         return false;
     }
+
+
+    @EventHandler
+    public boolean onShoot(ProjectileHitEvent e) {
+        if (e.getEntity() instanceof Arrow) {
+            Arrow arrow = (Arrow) e.getEntity();
+            Entity shooter = (Entity) arrow.getShooter();
+            Location loc=getLocation(e);
+            Bukkit.getServer().broadcastMessage(loc.toString());
+            if (arrow.hasCustomEffect(PotionEffectType.CONDUIT_POWER)) {
+                loc.getWorld().createExplosion(loc, 10, false, false, shooter);
```

Comment:
Issue #4, what do 10, false and false, represent, 10 arrows, 10 cows, does the first false mean don't damage anything? Make them into variables with names and we will deal with them later